### PR TITLE
dist,hack: add scripts to manage CI credentials

### PR DESCRIPTION
**Pull request by steveej**

### CODE REVIEW

File: dist/prepare_ci_credentials.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+set -xeuo pipefail
+
+test -f "${CINCINNATI_CI_DOCKERCFG_PATH}"
+test -f "${CINCINNATI_CI_PUBLIC_DOCKERCFG_PATH}"
+
+touch "${CINCINNATI_CI_DOCKERJSON_PATH}"
+touch "${CINCINNATI_CI_PUBLIC_DOCKERJSON_PATH}"
+
+# consolidate the quay credentials with the private CI registry credentials to allow a seamless transition to the latter
+jq -s '{ "auths": .[0] } * .[1]' "${CINCINNATI_CI_DOCKERCFG_PATH}" "${CINCINNATI_TEST_CREDENTIALS_PATH}" > "${CINCINNATI_CI_DOCKERJSON_PATH}"
```
Comment (LalatenduMohanty) [inline dist/prepare_ci_credentials.sh:14]: What is `${CINCINNATI_TEST_CREDENTIALS_PATH}"` ?

File: dist/prepare_ci_credentials.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+jq <"${CINCINNATI_CI_PUBLIC_DOCKERCFG_PATH}" '{ "auths": . }' > "${CINCINNATI_CI_PUBLIC_DOCKERJSON_PATH}"
```
Comment (LalatenduMohanty) [inline dist/prepare_ci_credentials.sh:15]: Without looking in to ` hack/get_ci_secrets.sh` it is hard to understand this script. Can you add a comment that the variables can be set running ` hack/get_ci_secrets.sh` for CI runs?

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: dist/prepare_ci_credentials.sh (hunk 2 above)
Comment (steveej) [reply dist/prepare_ci_credentials.sh:15]: > hack/get_ci_secrets.sh

That's only meant for local development to set up the environment that exists in CI. So on CI we don't want to run that hack script.

File: hack/get_ci_secrets.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+#
+# This script fetches the secrets from the CI cluster and populates local files with it.
+# This script is meant to be sourced.
```
Comment (LalatenduMohanty) [inline hack/get_ci_secrets.sh]: Can we mention that this file is mostly for the local development to set up the environment that exists in CI?

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: hack/get_ci_secrets.sh (hunk 3 above)
Comment (LalatenduMohanty) [reply hack/get_ci_secrets.sh]: In its current form it would be hard to comprehend with respect to its use by anyone who is new to Cincinnati.

Comment (steveej) [reply hack/get_ci_secrets.sh]: To me it's hard to comprehend how we would rely on anything in a directory called `hack` for anything other than local development environments :wink: I did change the description though. PTAL

**Comment by steveej** [comment]
Seems to be some unusual latency on the CI node.

/retest

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**